### PR TITLE
Add Docker image for Python Testing

### PR DESCRIPTION
**Pull request by khalludi**
Separates Docker test image code from PR #24034 . This allows us to test that the Docker test image is being built before it is used in the referenced PR.

This PR also partially takes care of the root issue #23830.

(@chaodaiG, please review at your convenience)

### CODE REVIEW

File: config/jobs/image-pushing/k8s-staging-test-infra.yaml [hunk 1]
```diff
@@ -273,3 +273,27 @@ postsubmits:
           - --project=k8s-staging-test-infra
           - --env-passthrough=PULL_BASE_REF
           - triage/
+    -name: post-test-infra-push-pytest
+      cluster: k8s-infra-prow-build-trusted
+      run_if_changed: '^(images/pytest|experiment|hack|kettle|metrics|releng)/'
+      annotations:
+        testgrid-dashboards: sig-testing-images, sig-k8s-infra-gcb
+        testgrid-tab-name: "pytest"
+        testgrid-alert-email: kubernetes-sig-testing-alerts@googlegroups.com, k8s-infra-alerts@kubernetes.io
```
Comment (chaodaiG) [inline config/jobs/image-pushing/k8s-staging-test-infra.yaml:282]: Let's not alert until we know it works

File: config/jobs/image-pushing/k8s-staging-test-infra.yaml [hunk 2]
```diff
@@ -273,3 +273,27 @@ postsubmits:
...
+        serviceAccountName: gcb-builder
+        containers:
+        - image: gcr.io/k8s-staging-test-infra/image-builder:v20211014-7ca1952a94
+          command:
+          - /run.sh
+          args:
+          - --scratch-bucket=gs://k8s-staging-test-infra-gcb
+          - --project=k8s-staging-test-infra
+          - --build-dir=.
+          - images/pytest
```
Comment (chaodaiG) [inline config/jobs/image-pushing/k8s-staging-test-infra.yaml:299]: add a new line at the end of the file, please also do this to the other files

File: images/pytest/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FROM python:3.8-slim-buster
+
+WORKDIR /app
+
+COPY metrics/get_jq.sh /app/get_jq.sh
+COPY requirements3.txt /app/requirements3.txt
+
+RUN /bin/bash -c "apt-get update && apt-get install -y make wget"
+RUN /bin/bash -c "pip3 install -r requirements3.txt"
```
Comment (chaodaiG) [inline images/pytest/Dockerfile:9]: When a PR updates `requirements3.txt`, do we expect `make pytests` also run `pip3 install -r requirements3.txt`?

**Review by chaodaiG — COMMENTED** [review]
/ok-to-test

File: config/jobs/image-pushing/k8s-staging-test-infra.yaml [hunk 4]
```diff
@@ -273,3 +273,27 @@ postsubmits:
           - --project=k8s-staging-test-infra
           - --env-passthrough=PULL_BASE_REF
           - triage/
+    -name: post-test-infra-push-pytest
```
Comment (chaodaiG) [inline config/jobs/image-pushing/k8s-staging-test-infra.yaml:276]: test failed, add a space between `-` and `name`

Comment (khalludi) [reply config/jobs/image-pushing/k8s-staging-test-infra.yaml]: Fixed!

**Comment by khalludi** [comment]
it seems like the 2 tests are failing because of py3_deps. Is there a way we can outright ignore this since we are deprecating it?

**Comment by chaodaiG** [comment]
We might not want to ignore them in this PR, at least not yet. I think the reason why the tests failed is because the changes in `requirements3.txt` file, one idea is revert the change in the file, and do `pip3 install ` in make rules for now. What do you think?

**Comment by khalludi** [comment]
It looks like all the tests are passing this time. Are we green to merge?

File: config/jobs/image-pushing/k8s-staging-test-infra.yaml (hunk 1 above)
Comment (chaodaiG) [reply config/jobs/image-pushing/k8s-staging-test-infra.yaml:282]: Please delete this line

File: images/pytest/Dockerfile (hunk 3 above)
Comment (chaodaiG) [reply images/pytest/Dockerfile:23]: Any idea?

**Review by chaodaiG — COMMENTED** [review]
Please take a look at my comments

File: config/jobs/image-pushing/k8s-staging-test-infra.yaml (hunk 1 above)
Comment (khalludi) [reply config/jobs/image-pushing/k8s-staging-test-infra.yaml:282]: Will do.

File: images/pytest/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+RUN /bin/bash -c "pip3 install certifi chardet google-api-core google-cloud-bigquery google-cloud-pubsub idna python-dateutil pytz requests six urllib3"
+RUN /bin/bash -c "chmod a+x get_jq.sh && ./get_jq.sh && mv jq-1.5 .."
+
+CMD ["make", "pytests"]
```
Comment (chaodaiG) [inline images/pytest/Dockerfile:27]: add a new line at the end of this file, as well as other files

File: requirements3.txt [hunk 6]
```diff
@@ -9,4 +9,4 @@ PyYAML==5.3
 ruamel.yaml==0.16.5
 setuptools==44.0.0
 sh==1.12.14
-singledispatch==3.4.0.3
+singledispatch==3.4.0.3
```
Comment (chaodaiG) [inline requirements3.txt:12]: I don't quite understand why this is considered a change, I'm afraid that there is some encoding thing going on, can you revert changes in this file?

**Comment by khalludi** [comment]
Seems like adding the new line at end of `requirement3.txt` reverted it back to what's in `master` branch appropriately. Also added `\n` to the end of all files.

**Review by chaodaiG — APPROVED** [review]
Thank you!

**Pull request merged**